packages/flutter/test/widgets/scroll_position_test.dart

```diff
@@ -140,6 +140,75 @@ Future<void> performTest(WidgetTester tester, bool maintainState) async {
   expect(find.text('100'), findsNothing, reason: 'with maintainState: $maintainState');
 }
 
+class ExpandingBox extends StatefulWidget {
+  const ExpandingBox({this.collapsedSize, this.expandedsize});
+
+  final double collapsedSize;
+  final double expandedsize;
+
+  @override
+  _ExpandingBoxState createState() {
+    return _ExpandingBoxState(collapsedSize: collapsedSize, expandedsize: expandedsize);
+  }
+}
+
+class _ExpandingBoxState extends State<ExpandingBox> with AutomaticKeepAliveClientMixin<ExpandingBox>{
+  _ExpandingBoxState({this.collapsedSize, this.expandedsize}) {
```

Comment:
the idiomatic thing here btw is to use `widget.collapsedSize` and for the `State` subclass to not have any constructor arguments.